### PR TITLE
Fix/select2 4.0 finding errors

### PR DESCRIPTION
**Pull request by kreintjes**
Fixes two find errors, see commit messages.

### CODE REVIEW

**Comment by Hirurg103** [comment]
Hi @kreintjes . Thank you for your PR. What problem does it solve?

**Comment by Hirurg103** [comment]
I looked at the commit messages and understand now. Could you add test for it?

**Comment by Hirurg103** [comment]
@kreintjes if you do not want to add tests you could give me permission to push to the fork and I will add the tests to your PR

**Comment by kreintjes** [comment]
Hi @Hirurg103. I tried to add specs (see commits). The spec for the Ambigous match fix works, however I couldn't get the spec for the ElementNotFound to work. It appears the within block is completely ignored from the spec in this project for some reason, but I'm not sure why. If you could take a look that would be great.

Also I see you're using capybara-webkit, which can only be used when installing some external dependencies (an outdated Qt 5.5 on macos). Maybe it might be better to switch to switch to Capybara's/Selenium's baked-in `selenium_chrome_headless` driver?

**Comment by Hirurg103** [comment]
Hi @kreintjes ! Thank you a lot for adding tests! I appreciate it!

> The spec for the Ambigous match fix works, however I couldn't get the spec for the ElementNotFound to work. It appears the within block is completely ignored from the spec in this project for some reason, but I'm not sure why

Yes, I know about this issue and faced it. I think that `within` overrides by some RSpec function. `I use Capybara.within {}` for now [see this code](https://github.com/Hirurg103/capybara_select2/blob/0.3.0/spec/selec2-v3/select_within_modal_spec.rb#L10-L12). I will debug it later and try to fix

> Also I see you're using capybara-webkit, which can only be used when installing some external dependencies (an outdated Qt 5.5 on macos). Maybe it might be better to switch to switch to Capybara's/Selenium's baked-in selenium_chrome_headless driver?

The original intention was to test that plugin works with capybara-webkit as we use it intensively in out internal projects

File: spec/select2_v4_spec.rb [hunk 1]
```diff
@@ -33,6 +33,15 @@
             expect(page).to have_css '.select2-container span', text: 'Wii'
           end
         end
+
+        context 'when searching for an option from within block' do
```
Comment (Hirurg103) [inline spec/select2_v4_spec.rb]: `context 'when searching for an option within a block'`

File: spec/select2_v4_spec.rb [hunk 2]
```diff
@@ -33,6 +33,15 @@
...
+          it 'should select an option from search results' do
+            within '#single' do
```
Comment (Hirurg103) [inline spec/select2_v4_spec.rb]: `Capybara.within`

**Comment by Hirurg103** [comment]
Hi @kreintjes thank you for contributing to this project!

**Pull request merged**